staging/src/k8s.io/kube-scheduler/config/v1alpha1/types.go

```diff
@@ -49,6 +50,17 @@ type KubeSchedulerConfiguration struct {
 	// HardPodAffinitySymmetricWeight represents the weight of implicit PreferredDuringScheduling affinity rule, in the range 0-100.
 	HardPodAffinitySymmetricWeight *int32 `json:"hardPodAffinitySymmetricWeight,omitempty"`
 
+	// TopologySpreadConstraints is the cluster-level configuration for
+	// topology-based spreading.
+	// Constraints are applied to pods that don't define any in their PodSpec.
+	// Label selectors must be empty, as they are deduced from the pods'
+	// membership in Services, Replication Controllers, Replica sets or Stateful
+	// sets.
+	// Omit to use the default: soft spreading among nodes and zones.
+	// Set to empty to disable.
+	// +listType=atomic
```

Comment:
Is it necessary? I rarely see this across API codebase.

BTW: usually the comments are identical with the part in `pkg/scheduler/apis/config/types.go`.

Reply:
listType seems to be a new requirement. If I don't put it, I need to add a violation here: https://github.com/kubernetes/kubernetes/blob/master/api/api-rules/violation_exceptions.list
see doc here: https://godoc.org/k8s.io/kube-openapi/pkg/idl#ListType

Made them identical.